forgerock-auth/src/main/java/org/forgerock/android/auth/devicebind/DeviceBindingStatus.kt

```diff
@@ -6,22 +6,53 @@
  */
 package org.forgerock.android.auth.devicebind
 
+import java.security.PrivateKey
+
 /**
  * State of the Device Binding errors
  */
 
-sealed class DeviceBindingStatus(var message: String, val clientError: String, val errorCode: Int? = null)
-data class Timeout(private val errorMessage: String = "Biometric Timeout", private val errorType: String = "Timeout"): DeviceBindingStatus(errorMessage, errorType)
-data class Abort(private val errorMessage: String = "User Terminates the Authentication",  private val errorType: String = "Abort", private val code: Int? = null): DeviceBindingStatus(errorMessage, errorType, code)
-data class Unsupported(private var errorMessage: String? = "Device not supported. Please verify the biometric or Pin settings", private val errorType: String = "Unsupported"):
-    DeviceBindingStatus(errorMessage ?: "Failed to generate keypair or sign the transaction", errorType)
-data class UnRegister(private val errorMessage: String = "PublicKey or PrivateKey Not found in Device", private val errorType: String = "Unsupported"): DeviceBindingStatus(errorMessage, errorType)
-object Success: DeviceBindingStatus("", "")
+sealed class DeviceBindingStatus<out T : Any>(var message: String?,
+                                              val clientError: String,
+                                              val errorCode: Int? = null)
+
+data class Timeout(private val errorMessage: String? = "Biometric Timeout",
+                   private val errorType: String = "Timeout",
+                   private val code: Int? = null) :
```

Comment:
Should we specify errorCodes and make sure those are aligned with iOS?

Reply:
We use enum or SealedClass to define the error. The "code" attribute is optional and it represents the Android internal error code during biometric authentication, the code is a platform-specific error.